spec/services/exports/export_donations_csv_service_spec.rb

```diff
@@ -2,23 +2,63 @@
   describe '#generate_csv_data' do
     subject { described_class.new(donation_ids: donation_ids).generate_csv_data }
     let(:donation_ids) { donations.map(&:id) }
+    let(:duplicate_item) do
+      FactoryBot.create(
+        :item, name: Faker::Appliance.equipment
+      )
+    end
+    let(:items_lists) do
+      [
+        [
+          [duplicate_item, 5],
+          [
+            FactoryBot.create(
+              :item, name: Faker::Appliance.equipment
+            ),
+            7
+          ],
+          [duplicate_item, 3]
+        ],
+        *(Array.new(3) do |i|
+          [[FactoryBot.create(
+            :item, name: Faker::Appliance.equipment
+          ), i + 1]]
+        end)
+      ]
+    end
+
+    let(:item_names) { items_lists.flatten(1).map(&:first).map(&:name).sort.uniq }
     let(:donations) do
       start_time = Time.current
-      Array.new(3) do |i|
-        create(
+
+      items_lists.each_with_index.map do |items, i|
+        # We're gonna mutate for dubious profit.
+        items = items.dup
+        # ¡MUTATE! 🎉
+        (item, quantity) = items.shift
```

Comment:
Not sure I really understand what these two lines are accomplishing - is there a reason we're duping the items? And I think this is overcomplicating it a bit - passing `item` into the donation `create` method isn't necessary. You can just create line items directly for all items in the array rather than separating out the first one from the rest of them. Unless I'm missing something this block is doing.

Reply:
If you don't provide an `item`, the factory will create a default one for you. And it doesn't have a built-in capability to create more than one item. I had added the `items` param to the changes to the factory that I subsequently rolled back to keep the factory simple.

Reply:
Doesn't it only do that if you use the `with_items` trait? You can just leave that off. :)

Also still not sure about the duping part?

Reply:
Ah yeah. Duh. :man_facepalming: 

Was again just following the pattern of what was there. Was thinking maybe some of the boilerplate in the factory was still necessary, but it's not. Now that I removed that, I was able to get rid of the `dup` + `shift`. 👍